x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -196,16 +219,19 @@ public void authenticationSuccess(String realm, User user, String action, Transp
             final Optional<String[]> indices = indices(message);
             if (eventFilterPolicyRegistry.ignorePredicate()
                     .test(new AuditEventMetaInfo(Optional.of(user), Optional.of(realm), Optional.empty(), indices)) == false) {
-                final LocalNodeInfo localNodeInfo = this.localNodeInfo;
+                final StringMapMessage logEntry = new StringMapMessage(this.entryCommonFields.commonFields);
+                logEntry.with(EVENT_TYPE_FIELD_NAME, TRANSPORT_ORIGIN_FIELD_VALUE)
+                        .with(EVENT_ACTION_FIELD_NAME, "authentication_success")
+                        .with(REALM_FIELD_NAME, realm)
+                        .with(ACTION_FIELD_NAME, action)
+                        .with(REQUEST_NAME_FIELD_NAME, message.getClass().getSimpleName());
+                principal(user, logEntry);
+                restOrTransportOrigin(message, threadContext, logEntry);
                 if (indices.isPresent()) {
-                    logger.info("{}[transport] [authentication_success]\t{}, {}, realm=[{}], action=[{}], indices=[{}], request=[{}]{}",
-                            localNodeInfo.prefix, originAttributes(threadContext, message, localNodeInfo), principal(user), realm, action,
-                            arrayToCommaDelimitedString(indices.get()), message.getClass().getSimpleName(), opaqueId());
-                } else {
-                    logger.info("{}[transport] [authentication_success]\t{}, {}, realm=[{}], action=[{}], request=[{}]{}",
-                            localNodeInfo.prefix, originAttributes(threadContext, message, localNodeInfo), principal(user), realm, action,
-                            message.getClass().getSimpleName(), opaqueId());
+                    logEntry.with(INDICES_FIELD_NAME, Strings.arrayToCommaDelimitedString(indices.get()));
```

Comment:
Should `indices` be an array instead of a comma delimited string?

Reply:
What will be the different in the log output? ["a,b,c"] vs ["a", "b", "c"]? If yes, prefer the second one as it does not need additional processing as I would say is as human readable as the first one.

Reply:
> What will be the different in the log output? ["a,b,c"] vs ["a", "b", "c"]?

Close, it is `"indices": "a,b,c"` instead of `"indices": ["a", "b", "c"]`.
There is some convenience of having it as `"a,b,c"`, because this is a string and not an array formatted as a string, but either one is doable! The trickiness lies at the `StringMapMessage` and `PatternLayout` interface.

Reply:
The advantage of `"indices": ["a", "b", "c"]` for us is that if we ingest it into Elasticsearch, it will be a keyword type that contains a list of 3 entries and a query for `b` will return a result. If we have `"indices": "a,b,c"` as keyword a query for `b` will not return a result if we don't split it up on Filebeat side or during ingest time.

Reply:
I have went with the `"indices": ["a", "b", "c"]` alternative. The same for `"user.roles"`.